File: 백준/Silver/3273. 두 수의 합/두 수의 합.cc
#include<iostream>
#include<algorithm>
#include<vector>

using namespace std;
 
int main() {
    int n, answer = 0, x;
    int left, right, tmp;
    
    cin >> n;
    int arr[n];

    for (int i = 0; i < n; i ++) {
        cin >> arr[i];
    }
    cin >> x;
    sort(arr, arr + n);

    left = 0;
    right = n - 1;

    while (left < right) {
        tmp = arr[left] + arr[right];
        if (tmp > x)
            right --;
        else if (tmp < x)
            left ++;
        else {
            answer ++;
            left ++;
        }
    }

    cout << answer;
}

File: 백준/Silver/3273. 두 수의 합/두 수의 합.py
n = int(input())
arr = list(map(int, input().split()))
tg_num = int(input())
result = 0

arr.sort()

left = 0
right = n - 1

while left < right:
    if arr[left] + arr[right] > tg_num:
        right -= 1
    elif arr[left] + arr[right] == tg_num:
        left += 1
        result += 1
    else:
        left += 1

print(result)